File: src/C#/BehavioralPatterns/Command/FastFoodRestaurant/Commands/RemoveMenuItemCommand.cs
using FastFoodRestaurant.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace FastFoodRestaurant.Commands
{
    public class RemoveMenuItemCommand : OrderCommand
    {
        public override void Execute(List<MenuItem> fastFoodOrderMenuItems, MenuItem menuItem)
        {
            var menuItemToRemove = fastFoodOrderMenuItems.FirstOrDefault(mi => mi.Name == menuItem.Name);

            if (menuItemToRemove != null)
            {
                fastFoodOrderMenuItems.Remove(menuItemToRemove);
            }
        }
    }
}

File: src/JavaScript/BehavioralPatterns/Command/FastFoodRestaurant/commands/removeMenuItemCommand.js

const OrderCommand = require('./orderCommand');

class RemoveMenuItemCommand extends OrderCommand {
    execute(fastFoodOrderMenuItems, menuItem) {
        const menuItemRemovalIndex = fastFoodOrderMenuItems.findIndex(mi => mi.name === menuItem.name);

        if (menuItemRemovalIndex !== -1) {
            fastFoodOrderMenuItems.splice(menuItemRemovalIndex, 1);
        }
    }
}

module.exports = RemoveMenuItemCommand;
